Editor/Gui/Graph/Interaction/ExampleSymbolLinking.cs

```diff
@@ -10,20 +12,62 @@ public static class ExampleSymbolLinking
         public static void UpdateExampleLinks()
         {
             ExampleSymbols.Clear();
-            foreach (var symbolUi in SymbolUiRegistry.Entries.Values)
+
+            const string exampleSuffix = "Example";
+            const string examplesSuffix = "Examples";
+
+            var potentialExamples = new List<(string correspondingSymbolName, Symbol exampleSymbol)>();
+
+            var symbolUiCollection = SymbolUiRegistry.Entries.Values;
+            Dictionary<string, Symbol> symbolsByName = new(capacity: symbolUiCollection.Count);
+            foreach (var symbolUi in symbolUiCollection)
+            {
+                var symbol = symbolUi.Symbol;
+                var name = symbol.Name;
+                
+                symbolsByName.Add(name, symbol);
+                
+                var endsWithExample = name.EndsWith(exampleSuffix);
+
+                if (endsWithExample)
+                {
+                    var correspondingSymbolName = name.Substring(0, name.Length - exampleSuffix.Length);
+                    potentialExamples.Add((correspondingSymbolName, symbol));
+                    continue;
+                }
+                
+                var endsWithExamples = name.EndsWith(examplesSuffix);
+
+                if (endsWithExamples)
+                {
+                    var correspondingSymbolName = name.Substring(0, name.Length - examplesSuffix.Length);
+                    potentialExamples.Add((correspondingSymbolName, symbol));
+                }
+            }
+
+            Log.Debug($"Found {potentialExamples.Count.ToString()} potential examples");
+            
+            foreach (var potentialExample in potentialExamples)
             {
-                var examples = SymbolUiRegistry.Entries.Values
-                                               .Where(c => c.Symbol.Name == symbolUi.Symbol.Name + "Example"
-                                                           || c.Symbol.Name == symbolUi.Symbol.Name + "Examples")
-                                               .Select(c => c.Symbol.Id)
-                                               .ToList();
-                if (examples.Count != 0)
-                    ExampleSymbols[symbolUi.Symbol.Id] = examples;
+                var gotSymbol = symbolsByName.TryGetValue(potentialExample.correspondingSymbolName, out var symbol);

```

Comment:
```suggestion
                if(!symbolsByName.TryGetValue(potentialExample.correspondingSymbolName, out var symbol)
                     continue;
```

Would be just fine for me.

Reply:
yeah I agree that's fine, I just think the former is more readable. six in one hand etc etc though